docs/logging.md

```diff
@@ -0,0 +1,86 @@
+# GCSFuse User Logging Guide
+Here, we will describe about the logging in gcsfuse and how different types of user
+can use the log-file flag to complete their work.
+
+GCSFuse supports different types of debug_flag named, --debug_fs, --debug_fuse, 
+--debug_gcs etc., which logs the debug info of different components in gcsfuse.
+
+## Log Location
+GCSFuse provides a flag named `--log-file` which controls the location of the logs.
+Either it can be a syslog or custom location. We will achieve the logging at
+stdout or stderr with the help of the same flag.
+
+We will also describe how we can achieve the rotation of logs in each case.
+
+### Default location - syslog
+By default, if you don't provide --log-file, GCSFuse writes all the logs to
+syslog which eventually redirects to `/var/log/gcsfuse.log`. The redirection happens
+with the help of rsyslog conf file, which you found at `/etc/rsyslog.d/08-gcsfuse.conf`,
```

Comment:
The redirection happens with the help of rsyslog conf file. rsyslog file is present at `/etc/rsyslog.d/08-gcsfuse.conf`, if GCSFuse is installed using the released-package. Otherwise, the configuration has to be added manually. e.g. in case when you mount by building
the source code.